### DISCUSSION TITLE
Zip Code should be optional

### DISCUSSION THREAD

**Discussion by seroujb** [woocommerce/woocommerce#49019] [Checkout Flow]
My country does not have Zip Code and i can not find a way to make it non-obligatory.

**Comment by PlatonB** [comment]
It is difficult to predict which checkout fields are relevant for specific countries. Therefore, **all** checkout fields except e-mail should be optional.

**Comment by mikejolley** [comment]
WooCommerce has a configuration file that defines which fields are required on a per-country basis. 

You can see how rules per country are defined here: https://github.com/woocommerce/woocommerce/blob/fcb7f6798f2ef9da5d6b48b8bc1fd3ed9452e413/plugins/woocommerce/includes/class-wc-countries.php#L885-L893

- You can customise fields yourself using a PHP filter like in this example: https://gist.github.com/alexminza/8384d53e4ef8e5c066c42bc559f15775
- You can create a bug report for a specific country here https://github.com/woocommerce/woocommerce/issues/new?assignees=&labels=&projects=&template=1-bug-report.yml so it can be fixed in plugin

> Therefore, all checkout fields except e-mail should be optional.

I don't think we should change the defaults because it will break existing use-cases. See my answer above.

**Reply by seroujb** [reply]
Thanks, I also found the code for that one country (LB) so I can make the zip code non mandatory but it would be nice to have an option in Woocommerce settings to customise without coding as not everyone can code and I think this is a simple feature. some Enable/Disable feature maybe (not to completely hide but make it Optional).

**Reply by mikejolley** [reply]
The goal would be to include all configs in the file I referenced in core, rather than introduce a new UI.

**Comment by mikejolley** [comment]
I am going to close this. I'd again encourage anyone who spots a country with incorrect address handling to raise an issue so it can be fixed in the locale for that country.

Thanks
